From 74060343a0286bf9d8054cbe67cb019d07455db3 Mon Sep 17 00:00:00 2001
From: Vlad Zahorodnii <vlad.zahorodnii@kde.org>
Date: Thu, 21 Mar 2024 10:36:11 +0200
Subject: [PATCH] tiles: Use deleteLater() in CustomTile::remove()

Otherwise it seems that QQmlEngine can segfault.

(cherry picked from commit ea9691a2f5a247235cbb8c3f326e4d08d1d821bb)
---
 src/tiles/customtile.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tiles/customtile.cpp b/src/tiles/customtile.cpp
index a8220184aba..9eba81ab345 100644
--- a/src/tiles/customtile.cpp
+++ b/src/tiles/customtile.cpp
@@ -320,7 +320,7 @@ void CustomTile::remove()
         }
     }
 
-    delete this;
+    deleteLater(); // not using "delete this" because QQmlEngine will crash
 }
 
 CustomTile *CustomTile::nextTileAt(Qt::Edge edge) const
-- 
GitLab

